gapic/src/main/com/google/gapid/models/Devices.java

```diff
@@ -142,6 +144,48 @@ public void selectReplayDevice(Device.Instance dev) {
     listeners.fire().onReplayDeviceChanged(dev);
   }
 
+  public void updateValidationStatus(DeviceCaptureInfo device, Service.ValidateDeviceResponse response) {
+    if (response == null || response.hasError()) {
+      device.validationStatus = false;
+      return;
+    }
+    device.validationStatus = true;
+  }
+
+  public void validateDevice(DeviceCaptureInfo device, Runnable callback) {
+    if (!devices.contains(device)) {
+      return;
+    }
+    rpcController.start().listen(client.validateDevice(device.path),
+        new UiErrorCallback<Service.ValidateDeviceResponse, Service.ValidateDeviceResponse, Service.ValidateDeviceResponse>(shell, LOG) {
+
+      @Override
+      protected void onUiThreadSuccess(Service.ValidateDeviceResponse response) {
+        updateValidationStatus(device, response);
+        callback.run();
+      }
+
+      @Override
+      protected void onUiThreadError(Service.ValidateDeviceResponse response) {
+        LOG.log(Level.WARNING, response.toString());
```

Comment:
nit: we typically statically import the levels

Reply:
Done.